apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt

```diff
@@ -1,10 +1,117 @@
+  fun <T : Any> adapterFor(scalarType: ScalarType): CustomTypeAdapter<T> {
+    var customTypeAdapter: CustomTypeAdapter<*>? = customAdapters[scalarType.typeName()]
+    if (customTypeAdapter == null) {
+      customTypeAdapter = DEFAULT_ADAPTERS[scalarType.className()]
+    }
+    return requireNotNull(customTypeAdapter) {
+      "Can't map GraphQL type: `${scalarType.typeName()}` to: `${scalarType.className()}`. Did you forget to add a custom type adapter?"
+    } as CustomTypeAdapter<T>
+  }
 
   companion object {
-    val DEFAULT: ScalarTypeAdapters
+    @JvmField
+    val DEFAULT = ScalarTypeAdapters(emptyMap())
+
+    private val DEFAULT_ADAPTERS = emptyMap<String, CustomTypeAdapter<*>>() +
+        createDefaultScalarTypeAdapter("java.lang.String", "kotlin.String") { value ->
+          if (value is GraphQLJsonList || value is GraphQLJsonObject) {
+            val buffer = Buffer()
+            JsonWriter.of(buffer).use { writer ->
+              Utils.writeToJson(value.value, writer)
+            }
+            buffer.readUtf8()
+          } else {
+            value.value.toString()
+          }
+        } +
+        createDefaultScalarTypeAdapter("java.lang.Boolean", "kotlin.Boolean") { value ->
+          when (value) {
+            is GraphQLBoolean -> value.value
+            is GraphQLString -> value.value.toBoolean()
+            else -> throw IllegalArgumentException("Can't decode: $value into Boolean")
+          }
+        } +
+        createDefaultScalarTypeAdapter("java.lang.Integer", "kotlin.Int") { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toInt()
+            is GraphQLString -> value.value.toInt()
+            else -> throw IllegalArgumentException("Can't decode: $value into Integer")
+          }
+        } +
+        createDefaultScalarTypeAdapter("java.lang.Long", "kotlin.Long") { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toLong()
+            is GraphQLString -> value.value.toLong()
+            else -> throw IllegalArgumentException("Can't decode: $value into Long")
+          }
+        } +
+        createDefaultScalarTypeAdapter("java.lang.Float", "kotlin.Float") { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toFloat()
+            is GraphQLString -> value.value.toFloat()
+            else -> throw IllegalArgumentException("Can't decode: $value into Float")
+          }
+        } +
+        createDefaultScalarTypeAdapter("java.lang.Double", "kotlin.Double") { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toDouble()
+            is GraphQLString -> value.value.toDouble()
+            else -> throw IllegalArgumentException("Can't decode: $value into Double")
+          }
+        } +
+        createDefaultScalarTypeAdapter(FileUpload::class.qualifiedName!!) { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toDouble()
```

Comment:
Why does this say Double for FileUpload?

Reply:
ogh, jeez my bad